### PR TITLE
full directory support for wiki

### LINKED ISSUE #823: Hierarchy (sub directory) in Wiki
### Feature Description
This Feature Request is about Support for Hierarchy/Subdirectories in Gitea Wiki. That includes below content.

- [ ] Files Can be Put in Subdirectories from UI or git client side
- [ ] Files In Subdirectories can be normally Visited from UI and git client side
- [ ] Files In Subdirectories can be normally Linked in Markdown to
- [ ] Subdirectories pages can be listed in the Pages View.
- [ ] Files Renamed in Sudirs stay in them. (API support for Subdirs)
- [ ] / in File names is interpreted as Path/folder even if it's in a Windows OS

### Why the feature is necessary

I think Support for Subdirectories is a Great Feature, because:
- It helps to bring better structure into a Wiki, especially for those big wiki
- It would work with a lot of Tools Out of the Box like: Obsidian and Logseq

### PR DESCRIPTION
**Pull request by Cherrg**
## directory support for wiki

previously this was part of #7225 - lets split it up

### This PR

* add support for **directories** inside wiki. Create/update/delete pages works well. If an old page is updated, the file will be moved to subdirectory.
  * allows to read files with nearly all letters, as it is possible to push all via wiki gitrepo
    * files can exists in old and new structure.
      * files are selected by link, so it is possible to access and view both
      * writing to one of them will delete both, and create new one in directory structure
      * delete will delete the selected file
   * **support images** relative to directory path
      * fix preview on preview rendering
   *  fix **500 error** if wiki contains pages with invalid uri encoding (e.g. filename: badescaping%%.md)
* ui fixes
  * add button 'Abort' to wiki `edit` and `new` page

fix #823

Signed-off-by: Michael Gnehr <michael@gnehr.de>

### CODE REVIEW

**Comment by lunny** [comment]
Please resolve the conflicts

File: models/wiki.go [hunk 1]
```diff
@@ -136,6 +198,9 @@ func (repo *Repository) updateWikiPage(doer *User, oldWikiName, newWikiName, con
 	}
 
 	gitRepo, err := git.OpenRepository(basePath)
+
+	fmt.Println(reflect.TypeOf(gitRepo))
```
Comment (lafriks) [inline models/wiki.go:202]: not needed

**Comment by lunny** [comment]
Please resolve the conflicts.

**Comment by gsantner** [comment]
@Cherrg

**Comment by ferrydeboer** [comment]
Well, this is frustrating, would really appreciate this feature. :( To bad I'm not proficient at Go.

**Comment by wxiaoguang** [comment]
This PR has been stale for long time, and the Wiki Title problem has been improved by https://github.com/go-gitea/gitea/pull/24143

Would there be new progress of this PR?

**Comment by wxiaoguang** [comment]
It has been stale for a long time and I can't think of a way to handling it other than closing it. Feel free to reopen if there's any new progress and I could also help.

**Pull request closed without merging**